### PR TITLE
[cmake] Get the default value for LLVM_ENABLE_ASSERTIONS from LLVMCon…

### PR DESCRIPTION
**Pull request by gottesmm**
#### Resolved bug number: ([SR-](https://bugs.swift.org/browse/SR-))

---

Before merging this pull request to apple/swift repository:
- [ ] Test pull request on Swift continuous integration.

<details>
  <summary>

Triggering Swift CI</summary>

The swift-ci is triggered by writing a comment on this PR addressed to the GitHub user @swift-ci. Different tests will run depending on the specific comment that you use. The currently available comments are:

**Smoke Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please smoke test |
| All supported platforms | @swift-ci Please smoke test and merge |
| OS X platform | @swift-ci Please smoke test OS X platform |
| Linux platform | @swift-ci Please smoke test Linux platform |

**Validation Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please test |
| All supported platforms | @swift-ci Please test and merge |
| OS X platform | @swift-ci Please test OS X platform |
| OS X platform | @swift-ci Please benchmark |
| Linux platform | @swift-ci Please test Linux platform |

**Lint Testing**

| Language | Comment |
| --- | --- |
| Python | @swift-ci Please Python lint |

Note: Only members of the Apple organization can trigger swift-ci.
</details>

…fig.cmake.

rdar://26154980

### CODE REVIEW

**Comment by gottesmm** [comment]
@swift-ci Please smoke test and merge

File: cmake/modules/SwiftSharedCMakeConfig.cmake [hunk 1]
```diff
@@ -162,10 +157,30 @@ macro(swift_common_standalone_build_config product is_cross_compiling)
...
+  # Then we import LLVMCONFIG_FILE. This is going to override whatever cached
+  # value we have for LLVM_ENABLE_ASSERTIONS.
   include(${LLVMCONFIG_FILE})
   set(LLVM_TOOLS_BINARY_DIR "${LLVM_TOOLS_BINARY_DIR_saved}")
-  set(LLVM_ENABLE_ASSERTIONS "${LLVM_ENABLE_ASSERTIONS_saved}")
+
+  # If we did not have a cached value for LLVM_ENABLE_ASSERTIONS, set
+  # LLVM_ENABLE_ASSERTIONS_saved to be the ENABLE_ASSERTIONS value from LLVM so
+  # we follow LLVMConfig.cmake's value by default if nothing is provided.
+  if (NOT LLVM_ENABLE_ASSERTIONS_saved)
```
Comment (jrose-apple) [inline cmake/modules/SwiftSharedCMakeConfig.cmake]: This treats "assertions explicitly disabled" and "assertions not set" the same, doesn't it?

Comment (gottesmm) [inline cmake/modules/SwiftSharedCMakeConfig.cmake]: Yes.

**Comment by gottesmm** [comment]
@swift-ci Please smoke test

**Pull request merged**